packages/lesswrong/components/hooks/useOrderPreservingArray.tsx

```diff
@@ -4,7 +4,8 @@ type OrderPreservingArrayPolicy = "prepend-new" | "append-new" | "interleave-new
 type IndexType = string | number;
 
 const arrayToIndexMap = (arr: IndexType[]): Record<IndexType, number> =>
-  Object.keys(arr).reduce(function (map, idx) {
+  Object.keys(arr).reduce(function (map, idxStr) {
+    const idx = parseInt(idxStr);
```

Comment:
`idx` was a string before, which surprisingly didn't cause any problems because when it was used for comparison in `.sort((a, b) => orderIndexMapRef.current[keyFunc(a)] - orderIndexMapRef.current[keyFunc(b)]);` javascript converts it to a number. But it's better for it to be explicit